translate/cloud-client/hybrid_glossaries/hybrid_tutorial.py

```diff
@@ -0,0 +1,257 @@
+    name = client.glossary_path(
+        project_id,
+        location,
+        glossary_name)
+
+    # Set language codes
+    language_codes_set = translate.types.Glossary.LanguageCodesSet(
+        language_codes=languages)
+
+    gcs_source = translate.types.GcsSource(
+        input_uri=glossary_uri)
+
+    input_config = translate.types.GlossaryInputConfig(
+        gcs_source=gcs_source)
+
+    # Set glossary resource information
+    glossary = translate.types.Glossary(
+        name=name,
+        language_codes_set=language_codes_set,
+        input_config=input_config)
+
+    resource = client.location_path(project_id, location)
+
+    # Create glossary resource
+    operation = client.create_glossary(parent=resource, glossary=glossary)
+
+    return operation.result(timeout=90)
+
+
+# [START hybrid_translate]
+def translate_text(text, prev_lang, new_lang, project_id, glossary_name):
+    # Translates text to a given language using a glossary
+    #
+    # ARGS
+    # text: String of text to translate
+    # prev_lang: language of input text
+    # new_lang: language of output text
+    # project_id: GCP project id
+    # glossary_name: name you gave your project's glossary
+    #  resource when you created it
+    #
+    # RETURNS
+    # String of translated text
+
+    # Instantiates a client
+    client = translate.TranslationServiceClient()
+
+    # Designates the data center location that you want to use
+    location = 'us-central1'
+
+    glossary = client.glossary_path(
+        project_id,
+        location,
+        glossary_name)
+
+    glossary_config = translate.types.TranslateTextGlossaryConfig(
+        glossary=glossary)
+
+    resource = client.location_path(project_id, location)
+
+    result = client.translate_text(
+        parent=resource,
+        contents=[text],
+        mime_type='text/plain',  # mime types: text/plain, text/html
+        source_language_code=prev_lang,
+        target_language_code=new_lang,
+        glossary_config=glossary_config)
+
+    # Returns translated text
+    return result.glossary_translations[0].translated_text
+    # [END hybrid_translate]
+
+
+# [START hybrid_glossary_delete]
+def delete_glossary(project_id, glossary_name):
+    # Deletes a GCP glossary resource
+    #
+    # ARGS
+    # project_id: GCP project id
+    # glossary_name: name you gave your project's glossary
+    #   resource when you created it
+    #
+    # RETURNS
+    # nothing
+
+    # Designates the data center location that you want to use
+    location = 'us-central1'
+
+    # Instantiates a client
+    client = translate.TranslationServiceClient()
+
+    resource = client.glossary_path(
+        project_id,
+        location,
+        glossary_name)
+
+    operation = client.delete_glossary(resource)
+    result = operation.result(timeout=90)
+
+    print('Deleted: {}'.format(result.name))
+    # [END hybrid_glossary_delete]
+
+
+# [START hybrid_tts]
+def text_to_speech(text, outfile):
```

Comment:
Since this code already exists, does it make since to re-use the existing methods and point your tutorial off to them?

Reply:
It's actually slightly different. The code I wrote for my previous tutorial uses SSML, not plaintext.